ADR/xxxx-kite-poc-architecture-and-components.md

```diff
@@ -0,0 +1,244 @@
+
+#### Application-Level Logic
+
+- **Upsert Pattern**: The system always checks for existing issues before creating new ones
+- **Duplicate Detection**: Matches issues based on:
+  - [Resource scope](https://github.com/konflux-ci/kite/blob/main/packages/backend/docs/Webhooks.md#benefits-of-scope-based-grouping) (type, name, namespace)
+  - Issue state (Active/Resolved)
+
+### Automatic Issue Lifecycle Management
+
+KITE implements automatic issue creation, updating and resolution using the combination of a custom controller + webhook. This minimizes manual intervention and prevents duplicate issue records.
+
+#### High-Level Issue Automation Overview
+
+This diagram shows a simplified flow of how KITE automatically detects and manages issues:
+
+```mermaid
+flowchart LR
+    subgraph "Kubernetes Cluster"
+        RESOURCE[Kubernetes Resource:<br/> PipelineRun, Deployment, etc.]
+        CONTROLLER[KITE Bridge Operator:<br/> Controllers Watch & Evaluate Kubernetes Resource]
+    end
+
+    subgraph "KITE Backend"
+        API[KITE API]
+        LOGIC[Issue Management: <br> Create/Update/Resolve]
+        DB[(PostgreSQL: <br/>Issue Storage)]
+    end
+
+    subgraph "User Interface"
+        DASHBOARD[Issues Dashboard]
+    end
+
+    %% Main Flow
+    RESOURCE -->|State Changes| CONTROLLER
+    CONTROLLER --> DECISION{Success or Failure?}
+
+    DECISION -->|Failure| FAIL_REQ[POST endpoint to upsert issue]
+    DECISION -->|Success| SUCCESS_REQ[POST endpoint to resolve active issues]
+
+    FAIL_REQ -->|Send event| API
+    SUCCESS_REQ -->|Send event| API
+
+    API --> LOGIC
+    LOGIC <--> DB
+    DB --> DASHBOARD
+```
+
+### Additional Architectural Decisions
+
+#### Modular Design
+
+- **Separate Packages**: Backend, CLI, and Operator are independent packages
+- **Clear Interfaces**: Well-defined APIs between components
+- **Independent Deployment**: Components can be deployed and scaled independently
+
+#### Configuration Management
+
+- **Environment Variables**: Extensive use of environment-based configuration
+- **Feature Flags**: Ability to enable/disable features like namespace checking and webhooks
+
+#### Development & Operations
+
+- **Container-First**: All components designed for containerized deployment
+- **Health Checks**: Built-in health endpoints for monitoring
+- **Logging**: Structured logging with configurable levels and formats
+- **Metrics**: Support for metrics collection (when enabled)
+
+## Consequences
+
+### Positive
+
+- **Scalability**: External database can handle large volumes of issue data while not overloading etcd
+- **Performance**: Database optimizations enable fast queries and reporting
+- **Extensibility**: Bridge operator pattern allows easy addition of new resource monitors
+- **Data Integrity**: Strong consistency guarantees prevent duplicate issues
+
+### Negative
+
+- **Complexity**: Additional infrastructure component (PostgreSQL) to manage
```

Comment:
And additional controllers and an additional stand-alone API. And there is no mention about how any of that is secured.

Reply:
> And additional controllers and an additional stand-alone API.

Yes good point, I'll add that.
> And there is no mention about how any of that is secured.

For the Backend API we do have a ticket to add a namespace check to ensure that a user has access to issues in a namespace: https://issues.redhat.com/browse/KFLUXUI-510

Are there any other security concerns you have?